pootle_pytest/factories.py

```diff
@@ -41,3 +45,112 @@ class AgreementFactory(factory.django.DjangoModelFactory):
 
     class Meta(object):
         model = 'staticpages.Agreement'
+
+
+class DirectoryFactory(factory.django.DjangoModelFactory):
+
+    @factory.lazy_attribute
+    def pootle_path(self):
+        if self.parent is None:
+            return "/"
+        return (
+            "%s/%s"
+            % (self.parent.pootle_path.rstrip("/"),
+               self.name))
+
+    class Meta:
+        model = 'pootle_app.Directory'
+        django_get_or_create = ("pootle_path", )
+
+
+class LanguageFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = 'pootle_language.Language'
+        django_get_or_create = ("code", )
+
+    code = factory.Sequence(lambda n: 'language%s' % n)
+    fullname = factory.Sequence(lambda n: 'Language %s' % n)
+
+
+class ProjectFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = 'pootle_project.Project'
+        django_get_or_create = ("code", )
+
+    code = factory.Sequence(lambda n: 'project%s' % n)
+    fullname = factory.Sequence(lambda n: 'Project %s' % n)
+    pootle_path = factory.LazyAttribute(lambda p: "/projects/%s" % p.code)
+
+
+class StoreFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = 'pootle_store.Store'
+        django_get_or_create = ("pootle_path", )
+
+    @factory.lazy_attribute
+    def pootle_path(self):
+        return ("%s/%s"
+                % (self.translation_project.pootle_path.rstrip("/"),
+                   self.name))
+
+    name = factory.Sequence(lambda n: 'store%s.po' % n)
+    parent = factory.LazyAttribute(
+        lambda s: s.translation_project.directory)
+
+
+class TranslationProjectFactory(factory.django.DjangoModelFactory):
+
+    class Meta:
+        model = 'pootle_translationproject.TranslationProject'
+
+
+def get_state_name(unit):
+    if unit.state == pootle_store.util.TRANSLATED:
+        return "translated"
+    elif unit.state == pootle_store.util.FUZZY:
+        return "fuzzy"
+    elif unit.state == pootle_store.util.OBSOLETE:
+        return "obsolete"
+    else:
```

Comment:
`else` is redundant here (let's avoid unnecessary nesting). Arguably the same applies to `elif`. But since this is just a mapping, you can also opt to use a simple dict instead:
```python
return {
  pootle_store.util.TRANSLATED: 'translated',
  pootle_store.util.FUZZY: 'fuzzy',
  pootle_store.util.OBSOLETE: 'obsolete',
}.get(unit.state, 'untranslated')
```